internal/app/grpc/delete_user_urls.go

```diff
@@ -0,0 +1,34 @@
+package grpc
+
+import (
+	"context"
+
+	"github.com/kirilltitov/go-shortener/internal/app/grpc/gen"
+	"github.com/kirilltitov/go-shortener/internal/logger"
+)
+
+func (a *Application) DeleteUserURLs(
+	ctx context.Context,
+	req *gen.DeleteUserURLsRequest,
+) (*gen.DeleteUserURLsResponse, error) {
+	userID, ok := getUserID(ctx)
+	if !ok {
+		return nil, ErrUnauthorized
+	}
+
+	doneCh := make(chan struct{})
+	a.wg.Add(1)
+	go func() {
+		defer a.wg.Done()
+
+		for err := range a.Shortener.DeleteUserURLs(context.Background(), doneCh, userID, req.UrlsToDel, a.wg) {
+			if err != nil {
+				logger.Log.Infof("Something went wrong during URL deletion: %s", err)
+			}
+		}
+
+		defer close(doneCh)
```

Comment:
```suggestion
		close(doneCh)
```
нет смысла последним оператором ставить отложенную функцию. Она и так будет выполнена в этой части кода.

Reply:
Это криво смержилось после предыдущего ревью :)